### PR TITLE
Remove UserListSerializer and UserListDeserializer from approvedBy field in ApprovalRule (fixes #589)

### LINKED ISSUE #589: UserListSerializer and UserListDeserializer should not be used on the approvedBy field of ApprovalRule
**Description of Problem**
`UserListSerializer` and `UserListDeserializer` are being applied to the `approvedBy` field of `ApprovalRule`. This is not correct, because this field is a normal list of users and does not need the special serializer and deserializer. You can see this is true by looking at the example from [GitLab's documentation](https://docs.gitlab.com/ee/api/merge_request_approvals.html#get-the-approval-state-of-merge-requests):
```
{
  "approval_rules_overwritten": true,
  "rules": [
    {
      "id": 1,
      "name": "Ruby",
      "rule_type": "regular",
      "eligible_approvers": [
        {
          "id": 4,
          "name": "John Doe",
          "username": "jdoe",
          "state": "active",
          "avatar_url": "https://www.gravatar.com/avatar/0?s=80&d=identicon",
          "web_url": "http://localhost/jdoe"
        }
      ],
      "approvals_required": 2,
      "users": [
        {
          "id": 4,
          "name": "John Doe",
          "username": "jdoe",
          "state": "active",
          "avatar_url": "https://www.gravatar.com/avatar/0?s=80&d=identicon",
          "web_url": "http://localhost/jdoe"
        }
      ],
      "groups": [],
      "contains_hidden_groups": false,
      "approved_by": [
        {
          "id": 4,
          "name": "John Doe",
          "username": "jdoe",
          "state": "active",
          "avatar_url": "https://www.gravatar.com/avatar/0?s=80&d=identicon",
          "web_url": "http://localhost/jdoe"
        }
      ],
      "source_rule": null,
      "approved": true,
      "overridden": false
    }
  ]
}
```
Note that `approved_by` is the exact same as `users`.

`UserListSerializer` and `UserListDeserializer` cause exceptions when trying to interact with merge request approval state JSON because of this mismatch.

**Suggested Fix**
The serializer and deserializer annotations should be removed from the `approvedBy` field of `ApprovalRule`.

### PR DESCRIPTION
**Pull request by tyates-indeed**
Remove UserListSerializer and UserListDeserializer from approvedBy field in ApprovalRule (fixes #589)

### CODE REVIEW

File: src/main/java/org/gitlab4j/api/models/ApprovalRule.java [hunk 1]
```diff
@@ -18,9 +18,6 @@
     private List<User> users;
     private List<Group> groups;
     private Boolean containsHiddenGroups;
-
-    @JsonSerialize(using = JacksonJson.UserListSerializer.class)
-    @JsonDeserialize(using = JacksonJson.UserListDeserializer.class)
```
Comment (derekyyyan) [inline src/main/java/org/gitlab4j/api/models/ApprovalRule.java:23]: Should also remove the unused imports.

Comment (tyates-indeed) [reply src/main/java/org/gitlab4j/api/models/ApprovalRule.java:23]: Yes, good catch. I have optimized imports with IntelliJ now.

**Comment by gmessner** [comment]
@tyates-indeed  and @derekyyyan 
The problem is that this will break the [Merge Request-level MR approvals](https://docs.gitlab.com/ee/api/merge_request_approvals.html#merge-request-level-mr-approvals)  

It is possible the documentation is incorrect, if it is correct then a bug issue should be filed with GitLab to make it consistent.

**Comment by tyates-indeed** [comment]
@gmessner all the unit tests passed which I assumed meant this would be okay. Are there not existing test cases for Merge Request-level MR approvals?

**Comment by gmessner** [comment]
@tyates-indeed Unfortunately, there are no existing unit tests for the Merge Request-level MR approvals which is why the unit tests passed.  I'll add unit tests for the Merge Request-level MR approvals  and then we figure out how to best address this inconsistency in the GitLab API

**Comment by gmessner** [comment]
@tyates-indeed After a thorough examination of the GitLab approval related API endpoints I have determined that this change is actually correct,  The calls that require the UserListerializer/Deserializer return an instance of ```MergeRequest``` which is properly configured, so this change to ```ApprovalRule``` is correct. I will be merging and releasing today.

**Comment by gmessner** [comment]
@tyates-indeed GitLab4J-API Release 4.15.3 has been published and is now available.

**Pull request merged**